storybook/src/components/TextInput/TextInput.docs.mdx

```diff
@@ -14,14 +14,116 @@ import README from "../../../../packages/css/src/components/text-input/README.md
+Use the password type when the input requires sensitive information, like passwords or PINs.
+It ensures that the input is not readable by others who might be looking at the screen.
+
+Guidelines:
+
+1. Allow the user’s password manager to automatically enter the password through `autocomplete="current-password"`.
+   When asking for a new password, use `autocomplete="new-password"` instead.
+2. Add a `minlength` attribute to ensure passwords meet a minimum length requirement.
+3. Use the `pattern` attribute to enforce password policies, like including numbers and special characters.
+4. If the password is a numeric PIN, add `inputmode="numeric"`.
+   Devices with virtual keyboards then switch to a numeric keypad layout which makes entering the password easier.
+
+<Canvas of={TextInputStories.Password} />
+
+### Type: Email address
+
+This field helps the user enter an email address.
+It has built-in validation to check for a valid email format.
+However, do not rely only on the browser’s validation.
+
+The `email` input field looks like a standard text input.
+On some devices, it may show an email-specific keyboard.
+
+Guidelines:
+
+1. Set `autocomplete="email"` to help browsers autofill the user’s email address.
+2. Use the `required` attribute to make sure the field is filled before the form is submitted.
+
+<Canvas of={TextInputStories.EmailAddress} />
+
+### Type: Web address
+
+This field helps the user enter an web address or URL.
+It has built-in validation to check for a valid format.
+However, do not rely only on the browser’s validation.
+
+The `url` input field looks like a standard text input.
+On some devices, it may show a URL-specific keyboard to aid in entering web addresses.
+
+<Canvas of={TextInputStories.WebAddress} />
+
+### Type: Phone number
+
+This field helps the user enter a phone number.
+It has built-in validation to check for a valid format.
+However, do not rely only on the browser’s validation.
+
+The `tel` input field looks like a standard text input.
+On mobile devices, it may display a numeric keypad or a keyboard optimized for entering phone numbers.
+
+Guidelines:
+
+1. Set `autocomplete="tel"` to help browsers autofill the user’s phone number.
+2. Use the `required` attribute to make sure the field is filled before the form is submitted.
+
+<Canvas of={TextInputStories.PhoneNumber} />
+
 ### Placeholder
 
+This text appears in the field when it is empty.
+
+Placeholder text can give a brief hint about the kind of data to enter.
+Use a word or short phrase that suggests the expected data.
+Do not use a full explanation or prompt.
+
+Do not use placeholders instead of a [Label](?path=/docs/components-forms-label--docs).
+Placeholders disappear when the field is filled.
+This makes it hard for the user to remember what they need to enter, reducing usability and accessibility.
+
 <Canvas of={TextInputStories.Placeholder} />
 
 ### Invalid
 
+Indicates that the input value does not meet the specified constraints.
```

Comment:
While adding types to `DateInput`, I find I could write exactly the same text for invalid and disabled field there. Should we extract it into a ‘Forms’ section in ‘Docs’ and link it there?

Reply:
The `The border of an invalid input is red`-part isn't true for all inputs, though, right? Switch doesn't have a invalid state, Radio has a red circle, etc.

Reply:
This page documents Text Inputs, not the other ones.

Reply:
Yeah, I meant: If we extract this section into a ‘Forms’ section in ‘Docs’, the visual description would not be correct for all inputs. 

I think we discussed Friday to not extract this now, and look at it again later?

Reply:
Correct.